distribution/test/itests/test-itests-alliance/src/test/java/org/codice/alliance/test/itests/NsiliEndpointTest.java

```diff
@@ -0,0 +1,242 @@
+import org.codice.alliance.test.itests.common.AbstractAllianceIntegrationTest;
+import org.codice.ddf.itests.common.annotations.AfterExam;
+import org.codice.ddf.itests.common.annotations.BeforeExam;
+import org.glassfish.grizzly.http.Method;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.osgi.service.cm.Configuration;
+
+import com.xebialabs.restito.semantics.Action;
+import com.xebialabs.restito.semantics.Call;
+import com.xebialabs.restito.semantics.Condition;
+import com.xebialabs.restito.server.StubServer;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class NsiliEndpointTest extends AbstractAllianceIntegrationTest {
+
+    private static final String[] REQUIRED_APPS =
+            {"catalog-app", "solr-app", "spatial-app", "nsili-app"};
+
+    private static final String CORBA_DEFAULT_PORT_PROPERTY =
+            "org.codice.alliance.corba_default_port";
+
+    private static final DynamicPort CORBA_DEFAULT_PORT = new DynamicPort(
+            CORBA_DEFAULT_PORT_PROPERTY,
+            6);
+
+    private static final DynamicPort RESTITO_STUB_SERVER_PORT = new DynamicPort(7);
+
+    private static final String NSILI_FILE_URI_PATH = "/nsili/file";
+
+    private static final DynamicUrl RESTITO_STUB_SERVER = new DynamicUrl("http://localhost:",
+            RESTITO_STUB_SERVER_PORT,
+            NSILI_FILE_URI_PATH);
+
+    private static final DynamicUrl NSILI_ENDPOINT_IOR_URL = new DynamicUrl(SERVICE_ROOT,
+            "/nsili/ior.txt");
+
+    private static final String SAMPLE_NSILI_CLIENT_FEATURE_NAME = "sample-nsili-client";
+
+    private static StubServer server;
+
+    private static String ingestedProductId;
+
+    @BeforeExam
+    public void beforeAllianceTest() throws Exception {
+        try {
+            basePort = getBasePort();
+            getAdminConfig().setLogLevels();
+
+            System.setProperty(CORBA_DEFAULT_PORT_PROPERTY, CORBA_DEFAULT_PORT.getPort());
+
+            getServiceManager().waitForRequiredApps(REQUIRED_APPS);
+            getServiceManager().waitForAllBundles();
+            getCatalogBundle().waitForCatalogProvider();
+
+            configureSecurityStsClient();
+
+            ingestedProductId = ingestRecord("alliance.png", "image/png");
+            startHttpListener();
+        } catch (Exception e) {
+            LOGGER.error("Failed in @BeforeExam: ", e);
+            fail("Failed in @BeforeExam: " + e.getMessage());
+        }
+    }
+
+    @AfterExam
+    public void stopHttpListener() {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    @Before
+    public void startSampleNsiliClientFeature() throws Exception {
+        getServiceManager().startFeature(true, SAMPLE_NSILI_CLIENT_FEATURE_NAME);
```

Comment:
After wrestling the with itest dependencies and consulting @kcwire, I decided that making a new sample client and restarting that feature for each test method was the best solution for now. Please, let me know if you know of a better solution.